diff --git a/widget/uikit/GeckoViewSwiftSupport.h b/widget/uikit/GeckoViewSwiftSupport.h
index 24cd4ecf0e62..2fb710721d4d 100644
--- a/widget/uikit/GeckoViewSwiftSupport.h
+++ b/widget/uikit/GeckoViewSwiftSupport.h
@@ -28,6 +28,8 @@
 @protocol SwiftGeckoViewRuntime <NSObject>
 - (id<SwiftEventDispatcher>)runtimeDispatcher;
 - (id<SwiftEventDispatcher>)dispatcherByName:(const char*)name;
+@optional
+- (void)childProcessDidStartWithPID:(int32_t)pid processType:(NSString*)processType;
 @end
 
 @protocol GeckoProcessExtension <NSObject>
