diff --git a/widget/uikit/nsLookAndFeel.h b/widget/uikit/nsLookAndFeel.h
index f491bd4e8bb8..f7205682ee62 100644
--- a/widget/uikit/nsLookAndFeel.h
+++ b/widget/uikit/nsLookAndFeel.h
@@ -13,6 +13,8 @@ class nsLookAndFeel final : public nsXPLookAndFeel {
   nsLookAndFeel();
   virtual ~nsLookAndFeel();
 
+  static void SetSystemUsesDarkTheme(bool aUseDarkTheme);
+
   void NativeInit() final;
   void RefreshImpl() override;
   nsresult NativeGetInt(IntID aID, int32_t& aResult) override;
@@ -28,8 +30,6 @@ class nsLookAndFeel final : public nsXPLookAndFeel {
   static bool UseOverlayScrollbars() { return true; }
 
  private:
-  nscolor mColorDarkText;
-
   bool mInitialized;
 
   void EnsureInit();
